src/app/components/collections/collections.component.ts

```diff
@@ -9,4 +11,32 @@ import { RouterLink } from '@angular/router';
   templateUrl: './collections.component.html',
   styleUrl: './collections.component.scss',
 })
-export class CollectionsComponent {}
+export class CollectionsComponent implements OnInit {
+  bookshelfId: number | undefined;
+  dataService = inject(DataService);
+  authService = inject(AuthService);
+  loggedInUserId: string | null | undefined = null;
+
+  ngOnInit() {
+    this.authService.isUserSet$.subscribe((user) => {
+      if (user) {
+        this.loggedInUserId = this.authService.currentUserSignal()?.id;
+      }
+    });
+
+    const bookshelfResult$ = this.dataService.getBookshelf(
+      this.loggedInUserId!,
```

Comment:
maybe just add `?? ''` for this since it is not guaranteed that loggedInUserId will be set when you do this, or add an if statement check